### ISSUE TITLE
Mirror So2Sat and implement download functionality

### ISSUE THREAD

**Issue by calebrob6** [microsoft/torchgeo#388]
Currently, the So2Sat dataset can't be auto downloaded because torchvision's `download_url` doesn't seem to support FTP URLs like "ftp://m1483140:m1483140@dataserv.ub.tum.de/training.h5".

The So2Sat authors are okay with having the dataset mirrored on a website like https://zenodo.org/. Once the dataset has been mirrored we can implement the download functionality as normal.

(@mehmetgunturkun this would be a great contribution if you have the time)

**Comment by calebrob6** [comment]
I haven't asked, however SEN12MS is 510GB so it wouldn't be trivial to upload to Zenodo (we'd have to split it up), and it would take a longg time to download over http. Maybe it is better to make users understand what they are getting into in that case?

**Comment by mehmetgunturkun** [comment]
Yes of course, it was the main reason why I asked! I'll take a look

**Comment by adamjstewart** [comment]
@wangyi111 and I talked about uploading these datasets to HF.
